src/targets/awsLambda.ts

```diff
@@ -0,0 +1,187 @@
+import { BaseArtifactProvider } from '../artifact_providers/base';
+import { ConfigurationError, reportError } from '../utils/errors';
+import { AWSError } from 'aws-sdk';
+import * as Lambda from 'aws-sdk/clients/lambda';
+import { PromiseResult } from 'aws-sdk/lib/request';
+
+const logger = loggerRaw.withScope(`[aws-lambda]`);
+
+/**
+ * RegExp for the AWS Lambda package.
+ * The pattern matches the following structure:
+ * `sentry-node-serverless-{version}.zip`.
+ */
+const DEFAULT_AWS_LAMBDA_DIST_REGEX = /sentry-node-serverless-\d+(\.\d+)*\.zip$/;
+
+const awsAllRegions = [
+  'us-east-1',
+  'us-east-2',
+  'us-west-1',
+  'us-west-2',
+  'ap-south-1',
+  'ap-southeast-1',
+  'ap-southeast-2',
+  'ap-northeast-1',
+  'ap-northeast-2',
+  'ca-central-1',
+  'eu-central-1',
+  'eu-west-1',
+  'eu-west-2',
+  'eu-west-3',
+  'sa-east-1',
+];
+
+const compatibleRuntimes = ['nodejs10.x', 'nodejs12.x'];
+
+/** Config options for the "aws-lambda" target. */
+interface AwsLambdaTargetOptions extends TargetConfig {
+  /** AWS access key ID, set as AWS_ACCESS_KEY_ID. */
+  awsAccessKeyId: string;
+  /** AWS secret access key, set as `AWS_SECRET_ACCESS_KEY`. */
+  awsSecretAccessKey: string;
+}
+
+/**
+ * The default layer name is used when no `AWS_LAYER_NAME`
+ * environment variable is found.
+ */
+export const defaultLayerName = 'SentryNodeServerlessSDK';
+
+/**
+ * Extracts the AWS Lambda layer name from the environment variables. If no
+ * environment variable is found, the default name is used.
+ */
+export function getAwsLayerName(): string {
+  if (!process.env.AWS_LAYER_NAME) {
+    return defaultLayerName;
+  } else {
+    return process.env.AWS_LAYER_NAME;
+  }
+}
+
+/**
+ * Target responsible for uploading files to AWS Lambda.
+ */
+export class AwsLambdaTarget extends BaseTarget {
+  /** Target name */
+  public readonly name: string = 'aws-lambda';
+  /** Target options */
+  public readonly awsLambdaConfig: AwsLambdaTargetOptions;
+  /** Name of the layer to be published */
+  static layerName: string;
+
+  public constructor(
+    config: TargetConfig,
+    artifactProvider: BaseArtifactProvider
+  ) {
+    super(config, artifactProvider);
+    this.awsLambdaConfig = this.getAwsLambdaConfig();
+    AwsLambdaTarget.layerName = getAwsLayerName();
+  }
+
+  /**
+   * Extracts AWS Lambda target options from the environment.
+   */
+  protected getAwsLambdaConfig(): AwsLambdaTargetOptions {
+    if (!process.env.AWS_ACCESS_KEY_ID || !process.env.AWS_SECRET_ACCESS_KEY) {
+      throw new ConfigurationError(
+        `Cannot perform AWS Lambda release: missing credentials.
+        Please use AWS_ACCESS_KEY_ID and AWS_SECRET_ACCESS_KEY environment variables.`
+      );
+    }
+    return {
+      awsAccessKeyId: process.env.AWS_ACCESS_KEY_ID,
+      awsSecretAccessKey: process.env.AWS_SECRET_ACCESS_KEY,
+    };
+  }
+
+  /**
+   * Publishes current lambda layer zip bundle to AWS Lambda.
+   * @param _version New version to be released.
```

Comment:
Why the underscore?

Reply:
It's an unused variable. AWS Lambda doesn't care about that version since it has it's own version counter for each layer ([available in the ARN](https://docs.aws.amazon.com/lambda/latest/dg/configuration-versions.html#versioning-versions-using)); where new versions are calculated on an auto-increment basis, starting from 1.

Reply:
Aha, I didn't notice it was unused... then why is it there? The examples in tests call either `.publish('version', 'revision')` or `.publish('', '')`, can't this be simply `publish(revision: string)`?

Reply:
The [method](https://github.com/getsentry/craft/blob/7bda0be6556047f7c39afa2107b5b0e2c368b2e0/src/targets/base.ts#L49) is overridden from the base class `BaseTarget`, which contains both parameters.